cynic-parser/src/ast/ids.rs

```diff
@@ -35,6 +48,21 @@ macro_rules! make_id {
                 &mut self.$field[(index.0.get() - 1) as usize]
             }
         }
+
+        impl IdRange<$name> {
+            pub fn len(&self) -> usize {
+                (self.end.0.get() - self.start.0.get()) as usize
+            }
+
+            pub fn is_empty(&self) -> bool {
+                (self.end.0.get() - self.start.0.get()) == 0
+            }
+
+            pub fn iter(&self) -> impl Iterator<Item = $name> {
+                (self.start.0.get()..self.end.0.get())
+                    .map(|num| $name(NonZeroU32::new(num).expect("range is too large")))
+            }
+        }
```

Comment:
Don't need this for _every_ id type - consider putting it in a separate macro, and only calling it where appropriate